KMPNativeCoroutinesAsync/AsyncFunction.swift

```diff
@@ -11,59 +11,31 @@ import KMPNativeCoroutinesCore
 /// - Parameter nativeSuspend: The native suspend function to await.
 /// - Returns: The result from the `nativeSuspend`.
 /// - Throws: Errors thrown by the `nativeSuspend`.
-public func asyncFunction<Result, Failure: Error, Unit>(for nativeSuspend: @escaping NativeSuspend<Result, Failure, Unit>) async throws -> Result {
-    let asyncFunctionActor = AsyncFunctionActor<Result, Unit>()
-    return try await withTaskCancellationHandler {
-        Task { await asyncFunctionActor.cancel() }
-    } operation: {
-        try await withUnsafeThrowingContinuation { continuation in
-            Task {
-                await asyncFunctionActor.setContinuation(continuation)
-                let nativeCancellable = nativeSuspend({ output, unit in
-                    Task { await asyncFunctionActor.continueWith(result: output) }
-                    return unit
-                }, { error, unit in
-                    Task { await asyncFunctionActor.continueWith(error: error) }
-                    return unit
-                })
-                await asyncFunctionActor.setNativeCancellable(nativeCancellable)
-            }
-        }
+public func asyncFunction<Result, Failure: Error, Unit>(
+    for nativeSuspend: @escaping NativeSuspend<Result, Failure, Unit>
+) async throws -> Result {
+    for try await result in asyncStream(for: nativeSuspend) {
+        return result
     }
+    try Task.checkCancellation()
```

Comment:
We are now actually just cancelling the suspend function in the background instead of waiting for it to cancel.